src/Microsoft.SqlTools.ManagedBatchParser/ReliableConnection/ReliableConnectionHelper.cs

```diff
@@ -869,6 +859,48 @@ public static ServerInfo GetServerVersion(IDbConnection connection)
             return result;
         }
 
+        private static void LookupClusterEndpoints(IDbConnection connection, ServerInfo serverInfo, List<ClusterEndpoint> clusterEndpoints)
+        {
+            try
+            {
+                ExecuteReader(
+                    connection,
+                    SqlConnectionHelperScripts.GetClusterEndpoints,
+                    delegate (IDataReader reader)
+                    {
+                        while (reader.Read())
+                        {
+                            clusterEndpoints.Add(new ClusterEndpoint {
+                                ServiceName = reader.GetString(0),
```

Comment:
So we have to also update all the calling code to correctly check all the options? Since new clusters will have these new properties filled out while old ones will have the old ones instead...we should definitely make sure we need to support older CTPs in this way and if so consider trying to make this easier to work with (do some parsing on our end here to fill in the IP/Port for example)

Reply:
Yes, the calling code is on the Azure Data Studio side. I've put in the compensating logic there and verified it still works when falling back to the old data. Need to verify things work with then new shape too 😄 Reason I'm not doing more on this side is that this should be short-term, after 1 month we can just remove the older code and not do compensation.